MySQL extension with MariaDB connector docs (#11608)
apache/druid: docs/development/extensions-core/mysql.md

To use this Apache Druid extension, make sure to [include](../../development/extensions.md#loading-extensions) `mysql-metadata-storage` as an extension.

> > The MySQL extension requires the MySQL Connector/J library or MariaDB Connector/J library, neither of which is not included in the Druid distribution.

Reviewer: ```suggestion
> The MySQL extension requires the MySQL Connector/J library or MariaDB Connector/J library, neither of which are included in the Druid distribution.
```
Double negative. Looks like neither are included.

> Refer to the following section for instructions on how to install this library.

## Installing the MySQL connector library

> This extension can use Oracle's MySQL JDBC driver which is not included in the Druid distribution and must be

Reviewer: ```suggestion
This extension can use Oracle's MySQL JDBC driver which is not included in the Druid distribution. You must```

> installed separately. There are a few ways to obtain this library:

Reviewer: ```suggestion
install it separately. There are a few ways to obtain this library:
```

Copy or symlink this file inside the folder `extensions/mysql-metadata-storage` under the distribution root directory.

## Alternative: Installing the MariaDB connector library

> THis extension also supports using the MariaDB connector jar, though it is also not included in the Druid distribution and must be installed separately like the MySQL connector.

Reviewer: ```suggestion
This extension also supports using the MariaDB connector jar. The MariaDB  connector is not included in the Druid distribution, so you must install it separately.
```

- Download from the MariaDB site: https://mariadb.com/downloads/connector

> - Maven Central: https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.7.3/mariadb-java-client-2.7.3.jar

Reviewer: ```suggestion
- Download from Maven Central: https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.7.3/mariadb-java-client-2.7.3.jar
```

> This should fetch a JAR file named similar to 'maria-java-client-x.x.x.jar'.

Reviewer: ```suggestion
This fetches the MariaDB connector JAR file with a name like 'maria-java-client-x.x.x.jar'.
```

Copy or symlink this file to `extensions/mysql-metadata-storage` under the distribution root directory.

> In Druid properties, `druid.metadata.mysql.driver.driverClassName=org.mariadb.jdbc.Driver` must be set, which will instruct the `mysql-metadata-storage` extension to use the MariaDB connector library instead of MySQL. Depending on the MariaDB client library version, it will support both `jdbc:mysql:` and `jdbc:mariadb:` connection URIs, though the parameters to configure the connection vary between implementations, so be sure to [check the documentation](https://mariadb.com/kb/en/about-mariadb-connector-j/#connection-strings) for details.

Reviewer: ```suggestion
To configure the `mysql-metadata-storage` extension to use the MariaDB connector library instead of MySQL,  set the following in the Druid properties: `druid.metadata.mysql.driver.driverClassName=org.mariadb.jdbc.Driver` 
Depending on the MariaDB client library version, the connector supports both `jdbc:mysql:` and `jdbc:mariadb:` connection URIs. However, the parameters to configure the connection vary between implementations, so be sure to [check the documentation](https://mariadb.com/kb/en/about-mariadb-connector-j/#connection-strings) for details.
```

  Alternatively, download and follow installation instructions for MySQL
  Community Server here:
  [http://dev.mysql.com/downloads/mysql/](http://dev.mysql.com/downloads/mysql/).

> This extension should also function correctly using MariaDB server, https://mariadb.org/download/, substituting for MariaDB in the following instructions where appropriate.

Reviewer: ```suggestion
This extension should also supports MariaDB server, https://mariadb.org/download/, substituting for MariaDB in the following instructions where appropriate.
```

  druid.metadata.storage.connector.password=diurd
  ```

> If using MariaDB, `druid.metadata.mysql.driver.driverClassName=org.mariadb.jdbc.Driver` must also be set.

Reviewer: ```suggestion
If using MariaDB, set `druid.metadata.mysql.driver.driverClassName=org.mariadb.jdbc.Driver`.
```
